### PR TITLE
Package and decorate error in memoizer for better general error logging

### PR DESCRIPTION
**Pull request by jhsware**
We could improve error handling in the actual resolver that causes the error, but this is the general error handler and will do for now.

### CODE REVIEW

File: kpm/kpm-backend/src/api/commonCache.ts [hunk 1]
```diff
@@ -62,13 +62,12 @@ export function memoized<T extends TCacheable>({
...
     // Cache miss, calculate value, store and return
-    let newValue: T | undefined;
-    try {
-      newValue = await fn(key);
-    } catch (err: any) {
-      Error.captureStackTrace(err);
-      log.error(err, "Missing error handling in memoized function");
-    }
+    const newValue: T | undefined = await fn(key).catch((err: any) => {
+      err.details = { key };
```
Comment (kaj) [inline kpm/kpm-backend/src/api/commonCache.ts:66]: Hm.  What if `err` has other details?  Mabye either just set `err.memoized_key = key` to make conflicts less probable (and avoid a single-field dictionary), or do something like:

```ts
   err.details |= {};
   err.details["memoized_key"] = key;
```

?

**Review by kaj — COMMENTED** [review]
Looks good.  I have one question:

File: kpm/kpm-backend/src/api/commonCache.ts (hunk 1 above)
Comment (jhsware) [reply kpm/kpm-backend/src/api/commonCache.ts]: Fixed in e08918a05

**Review by kaj — APPROVED** [review]
LGTM.  Som vanligt tar du mitt ändringsförslag och gör som jag säger fast lite snyggare.  😎

**Pull request merged**